src/KafkaFlow.OpenTelemetry/OpenTelemetryConsumerObserver.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace KafkaFlow.OpenTelemetry
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Text;
+    using System.Threading.Tasks;
+    using global::OpenTelemetry;
+    using global::OpenTelemetry.Context.Propagation;
+
+    internal static class OpenTelemetryConsumerObserver
```

Comment:
Perhaps the name 'Observer' may not be ideal as it has the potential to create confusion with the Observer pattern.